pulse/product-search-with-typesense/README.md

```diff
@@ -0,0 +1,223 @@
+- A [**Typesense Admin API key**](https://typesense.org/docs/0.20.0/api/api-keys.html#create-an-api-key) that allows you to perform all operations and is used in the backend. Configuration to create the Admin API key
+    ```json
+    {
+      "description": "Admin key.",
+      "actions": [
+        "*"
+      ],
+      "collections": [
+        "*"
+      ]
+    }
+    ```
+- A [**Typesense Search API Key**](https://typesense.org/docs/0.20.0/api/api-keys.html#create-an-api-key) that allows you to search products from the frontend. This API key is exposed on the frontend and allows you to perform a scoped search on the `products` collection. Configuration to create the Search API key:
+    ```json
+    {
+      "description": "Search-only product key.",
+      "actions": [
+        "documents:search"
+      ],
+      "collections": [
+        "products"
+      ]
+    }
+    ```
+
+## Getting started
+
+### 1. Clone the respository
+
+Clone the repository and navigate into the example:
+
+```
+git clone git@github.com:prisma/prisma-examples.git --depth=1
+cd prisma-examples/pulse/typesense-product-search
+```
+
+### 2. Configure and start the backend
+
+For the example to work, we have to setup and run the backend first. Navigate to the backend folder and install dependencies:
+
+```bash
+cd backend
+npm install
+```
+
+#### 1. Configure environment variables for the backend
+
+Create a `.dev.vars` in the root of the project directory:
+
+```bash
+touch .dev.vars
+```
+
+Now, open the `.dev.vars` file and update the `DATABASE_URL`, `TYPESENSE_ADMIN_API_KEY`, `TYPESENSE_URL`  and `PULSE_API_KEY` environment variables with the values of your connection string, your Pulse, your Typesense server URL and Typesense API key:
+
+```bash
+# .dev.vars
+DATABASE_URL=""
+PULSE_API_KEY=""
+TYPESENSE_ADMIN_API_KEY=""
+# Typesense URL should look like `https://xxx.typesense.net` (if you're using Typesense Cloud) or your self-hosted Typesense URL
+TYPESENSE_URL=""
+```
+
+> Note that the API keys **must** be filled in for the app to work.
+
+#### 2. Run a database migration to create the `Product` table
+
+The [Prisma schema file](./backend/prisma/schema.prisma) in this project contains a single `Product` model. You can map this model to the database and create the corresponding `Product` table using the following command:
+```bash
+npm run migrate
+```
+
+You now have a table called `Product` in your database.
+
+Next, run the following command to [seed](./backend/prisma/seed.js) the database with some dummy data:
+```bash
+npx prisma db seed
+```
+
+The is invoked the [seed script](./prisma/seed.js) and created ten `Product` records in the database.
+
+
+#### 3. Start the backend server
+
+Make sure you're inside the [`backend`](./backend/) directory and start the server that periodically syncs data to Typesense:
+
+```bash
+npm run dev
+```
+
+The server will start up at `http://localhost:8787`.
+
+#### 4. Test the CRON task
```

Comment:
```suggestion
#### 2.1. Configure environment variables for the backend

Create a `.dev.vars` in the root of the project directory:

```bash
touch .dev.vars
```

Now, open the `.dev.vars` file and update the `DATABASE_URL`, `TYPESENSE_ADMIN_API_KEY`, `TYPESENSE_URL`  and `PULSE_API_KEY` environment variables with the values of your connection string, your Pulse, your Typesense server URL and Typesense API key:

```bash
# .dev.vars
DATABASE_URL=""
PULSE_API_KEY=""
TYPESENSE_ADMIN_API_KEY=""
# Typesense URL should look like `https://xxx.typesense.net` (if you're using Typesense Cloud) or your self-hosted Typesense URL
TYPESENSE_URL=""
```

> Note that the API keys **must** be filled in for the app to work.

#### 2.2. Run a database migration to create the `Product` table

The [Prisma schema file](./backend/prisma/schema.prisma) in this project contains a single `Product` model. You can map this model to the database and create the corresponding `Product` table using the following command:
```bash
npm run migrate
```

You now have a table called `Product` in your database.

Next, run the following command to [seed](./backend/prisma/seed.js) the database with some dummy data:
```bash
npx prisma db seed
```

The is invoked the [seed script](./prisma/seed.js) and created ten `Product` records in the database.


#### 2.3. Start the backend server

Make sure you're inside the [`backend`](./backend/) directory and start the server that periodically syncs data to Typesense:

```bash
npm run dev
```

The server will start up at `http://localhost:8787`.

#### 2.4. Test the CRON task
```